src/main/java/com/google/sps/servlets/SpecificGuideServlet.java

```diff
@@ -0,0 +1,55 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.gson.Gson;
+import com.google.sps.data.Error;
+import com.google.sps.data.Guide;
+import com.google.sps.storage.DatastoreGuides;
+
+@WebServlet("/guides/*")
+public class SpecificGuideServlet extends HttpServlet{
+    static final long serialVersionUID = 0;
+    private final DatastoreGuides helper = new DatastoreGuides();
+    private final Gson gson = new Gson();
+    private long id;
+
+    private Optional<Error> getIdFromPath(String path){
+        String[] parts = path.split("/");
+        try{
+            id = Long.parseLong(parts[1]);
```

Comment:
Passing values this way doesn't work well with concurrency. You don't really need it here; instead, return the int and throw the exception.